test/blackbox-tests/test-cases/github5264.t

```diff
@@ -0,0 +1,49 @@
+Reproduction case for #5264
+
+Before Dune 3.0, it was possible to abuse the
+%{lib:<libname>:<filename>} syntax to refer to the library directory
+by writing %{lib:<libname:} or %{lib:<libname>:.}. Several projects
+used that in their ctypes rules. Dune 3.0 initially broke this, which
+caused many build failures. This test makes sure we still support this
+syntax in lang Dune < 3.0.
+
+  $ cat > dune <<'EOF'
+  > (library (name ctypes))
+  > (rule
+  >  (alias a)
+  >  (action (run echo %{lib:ctypes:})))
+  > (rule
+  >  (alias b)
+  >  (action (run echo %{lib:ctypes:.})))
```

Comment:
Maybe use a different name here instead of `ctypes` to test the other error as well.

Reply:
Sounds good, I did it and found an issue with the test at the same time! (I was using `lib` with a private library, while I should have been using `lib-private`).